onnxruntime/lora/adapter_format_utils.cc

```diff
@@ -75,35 +77,82 @@ const Adapter* ValidateAndGetAdapterFromBytes(gsl::span<const uint8_t> bytes) {
   return adapter;
 }
 
+template <class T>
+struct WriteDataForLittleEndian {
+  Status operator()(gsl::span<const uint8_t> src, gsl::span<uint8_t> dest) const {
+    auto src_span = ReinterpretAsSpan<const T>(src);
+    auto dest_span = ReinterpretAsSpan<unsigned char>(dest);
+    return onnxruntime::utils::WriteLittleEndian<T>(src_span, dest_span);
+  }
+};
+
 void SaveLoraParameter(flatbuffers::FlatBufferBuilder& flat_builder, std::string_view name,
                        TensorDataType data_type, gsl::span<const int64_t> shape,
                        gsl::span<const uint8_t> data,
                        flatbuffers::Offset<Parameter>& fbs_tensor) {
   auto name_str = (name.empty()) ? 0 : flat_builder.CreateString(name.data(), name.size());
   auto shape_vec = flat_builder.CreateVector(shape.data(), shape.size());
-  auto data_vec = flat_builder.CreateVector(data.data(), data.size());
 
+  flatbuffers::Offset<flatbuffers::Vector<uint8_t>> data_vec;
+  if constexpr (endian::native == endian::big) {
+    onnxruntime::utils::MLTypeCallDispatcher<float, double, int8_t, uint8_t,
+                                             int16_t, uint16_t, int32_t, uint32_t,
+                                             int64_t, uint64_t,
+                                             BFloat16, MLFloat16>
+        disp(static_cast<int32_t>(data_type));
+
+    InlinedVector<uint8_t> be_data(data.size());
+    auto status = disp.InvokeRet<Status, WriteDataForLittleEndian>(data, be_data);
+    ORT_THROW_IF_ERROR(status);
+    data_vec = flat_builder.CreateVector<uint8_t>(be_data.data(), be_data.size());
+  } else {
+    data_vec = flat_builder.CreateVector(data.data(), data.size());
+  }
   fbs_tensor = CreateParameter(flat_builder, name_str, shape_vec, data_type, data_vec);
 }
 
+template <class T>
+struct ReadDataForBigEndian {
+  Status operator()(gsl::span<const uint8_t> src, Tensor& dst) const {
+    auto src_span = ReinterpretAsSpan<const unsigned char>(src);
```

Comment:
nit: probably doesn't matter but could we do the data type conversion from uint8_t to unsigned char outside of the templatized code?